### PR TITLE
FIX Don't mark mocked response as an error

### PR DESCRIPTION
**Pull request by GuySartorelli**
This response type is only ever returned from the `findgridfield` action on the `GridFieldQueuedExportButton` (see `GridFieldQueuedExportButton::handleAction()`). The only way this action is accessed in normal operation is from the `GenerateCSVJob::getGridField()` method, which uses a _mock request_ via `Director::test()` to get the response. More discussion about why this happens is in the parent issue.

In other words, the only way this response class is ever used, it _cannot_ be a 500 error. It's always going to be a success - unless some exception is thrown, in which case this response won't be used anyway.

My best guess as to why this was set to `500` is in case some smart-alek decides to manually access the `findgridfield` action by typing in the appropriate URL to do so.... which is both a super weird edge case and also not something that explicitly needs a 500 response. A 200 response is, I'd argue, more appropriate even in that scenario, since the action _has_ done what it's meant to do. There was no error.

tl;dr: `500` was probably never correct for this in the first place.

## Issue
- https://github.com/silverstripe/silverstripe-gridfieldqueuedexport/issues/76

### CODE REVIEW

**Review by emteknetnz — APPROVED** [review]

**Comment by emteknetnz** [comment]
Tested locally, works good. The 500 in the constructor looks really weird and I cannot think of any reason to keep it, so happy to remove.
